crypto/rand/rand_lib.c

```diff
@@ -547,6 +541,13 @@ unsigned char *rand_pool_detach(RAND_POOL *pool)
     return ret;
 }
 
+/*
+ * Re-attach the |pool| buffer.
+ */
+void rand_pool_reattach(RAND_POOL *pool, unsigned char *buffer)
+{
+    pool->buffer = buffer;
+}
```

Comment:
The buffer should be cleared and the entropy count reset when it is reattached:
```diff
diff --git a/crypto/rand/rand_lib.c b/crypto/rand/rand_lib.c
index 78d0e304ab..f352486601 100644
--- a/crypto/rand/rand_lib.c
+++ b/crypto/rand/rand_lib.c
@@ -547,6 +547,8 @@ unsigned char *rand_pool_detach(RAND_POOL *pool)
 void rand_pool_reattach(RAND_POOL *pool, unsigned char *buffer)
 {
     pool->buffer = buffer;
+    OPENSSL_cleanse(pool->buffer, pool->len);
+    pool->entropy = 0;
 }
```

Reply:
Okay, currently pool->len and pool->entropy are cleared in rand_pool_detach,
but I can move those two statements to rand_pool_reattach.

Reply:
Of course it wouldn't make sense to clear pool->len on reattach ;-)

BTW: maybe it would be better to leave rand_pool_detach() as it is, but pass the length back into rand_pool_reattach(). This would be also more consistent with the rest the api.

Reply:
No I think I disagree, because rand_pool_add uses len as a cursor,
and the pool need to be in the original state.  max_len is probably what you meant?
But I like you suggestion to keep the len intact, and reset that only after the buffer is
reattached.

Reply:
okay, currently (but not in the original version) you can no longer use
rand_pool_length(pool); after   *pout = rand_pool_detach(pool);
But keeping len and/or entropy intact in rand_pool_detach is probably
an option.
I think I wanted to clear len in the rand_pool_detach code, to avoid
the get_entropy to succeed if it is called a second time with the static
buffer attached.
if rand_pool_detach clears entropy that prevents 
entropy_available = rand_pool_acquire_entropy(pool);
from returning nonzero entropy again, I mean in the unlikely case
that get_entropy is called twice with the same seed buffer.

Reply:
> max_len is probably what you meant?

Yes, my fault.

Reply:
OK, then let's go with max_len.

Reply:
Oh, sorry, that won't work because max_len = 12288, and len = 16,
cleansing 12K eats exactly 100% of the perfomance improvement.